app/models/concerns/content_annotatable.rb

```diff
@@ -0,0 +1,52 @@
+module ContentAnnotatable
+  extend ActiveSupport::Concern
+
+  included do
+    has_one :raw_content, as: :source, dependent: :destroy
+    has_many :resources, class_name: 'Content::Resource', as: :resource
+    has_many :annotations, through: :resources
+    before_create :prepare_raw_content, :sanitize_content
+    before_update :sanitize_content, if: :content_changed?
+    after_update :update_annotation_offsets, if: [:saved_change_to_content?, :annotated?]
+  end
+
+  class_methods do
+    def annotated
+      where("annotations_count > 0")
+    end
+  end
+
+  def annotated?
+    annotations_count > 0
+  end
+
+  private
+
+  def prepare_raw_content
+    build_raw_content(content: content)
+    true
+  end
+
+  def sanitize_content
+    self.content = HTMLUtils.sanitize(content)
+    true
+  end
+
+  def update_annotation_offsets
+    diffs = Differ.get_diffs(HTMLUtils.parse(previous_changes[:content][0]).text,
+                             HTMLUtils.parse(previous_changes[:content][1]).text)
```

Comment:
Could you make these more explicit that [0] is the previous content and [1] is the new value?